app/importers/protege_importer.rb

```diff
@@ -0,0 +1,180 @@
+require 'open-uri'
+require 'zip'
+
+class ProtegeImporter
+
+  # This is the STAGING protege URL, will need to switch this to Production
+  ZIP_URL = "http://52.4.82.207:8080/webprotege/download?ontology=0aa09276-58a6-4350-a0bb-60eb2ab4be00"
+
+  def initialize(resource = ZIP_URL)
+    @resource = resource
+    @root_term = {}
+    @all_terms, @industry_terms, @industries, @sectors = [], [], [], []
+  end
+
+  def import
+    xml = extract_xml_from_zip
+    parse_terms_from_xml(xml)
+
+    @all_terms.each { |term| @industry_terms.push(term) if validate_industries_term(term) }
+    extract_industries_and_sectors
+    @sectors = find_parent_industries
+
+    update_or_add_industries
+    update_or_add_sectors
+    delete_old_terms
+  end
+
+  def delete_old_terms
+    Industry.all.each do |db_industry|
+      if @industries.find { |protege_industry| protege_industry[:protege_id] == db_industry.protege_id } == nil
+        db_industry.destroy
+      end
+    end
+
+    Sector.all.each do |db_sector|
+      if @sectors.find { |protege_sector| protege_sector[:protege_id] == db_sector.protege_id } == nil
+        db_sector.destroy
+      end
+    end
+  end
+
+  def update_or_add_industries
+    @industries.each do |term|
+      if Industry.exists?(protege_id: term[:protege_id])
+        industry = Industry.find_by(protege_id: term[:protege_id])
+        industry.update(name: term[:name])
+      else
+        industry = Industry.create(protege_id: term[:protege_id], name: term[:name])
+      end
+
+    end
+  end
+
+  def update_or_add_sectors
+    @sectors.each do |term|
+      term[:industry_ids].each do |industry_id|  
+        industry = Industry.find_by(protege_id: industry_id)
+        if industry.sectors.exists?(protege_id: term[:protege_id])
+          sector = industry.sectors.find_by(protege_id: term[:protege_id])
+          sector.update(name: term[:name])
+        else
+          sector = industry.sectors.create(protege_id: term[:protege_id], name: term[:name])
+        end
+      end
+    end
+  end
+
+  def extract_industries_and_sectors
+    @industry_terms.each do |term|
+      if term[:parent_ids] == [@root_term[:protege_id]]
+        @industries.push(term)
+      else
+        @sectors.push(term.merge({industry_ids: []}))
+      end
+    end
+  end
+
+  def validate_industries_term(term)
+    if term[:parent_ids].include? @root_term[:protege_id]
+      return true
+    elsif ['skos:Concept', 'Concept Scheme', 'Collection'].include?(term[:name]) || term[:parent_ids] == []
+      return false
+    else
+      @all_terms.each do |t|
+        return validate_industries_term(t) if term[:parent_ids].include?(t[:protege_id])
+      end
+    end
+  end
```

Comment:
For methods that return only true or false, it's a common naming convention to put a "?" at the end (the rationale is that you are asking a yes/no question). A better name for this would be `industry_term_valid?` (Also, it's unclear to me whether you're passing a single term or multiple terms. Either way, I would have thought "industry_term" or "industry_terms" would be more grammatically correct than "industries_term". Maybe I don't understand the code well enough though.)